projects/views.py

```diff
@@ -828,6 +837,52 @@ def edit_type(request, project_id):
     return form
 
 
+@require_POST
+def edit_pi(request, project_id):
+    if not request.user.is_superuser:
+        messages.error(request, "Only the admin users can update project PI.")
+        return EditPIForm(request.POST)
+
+    form = EditPIForm(request.POST)
+    if form.is_valid():
+        project_pi_username = form.cleaned_data["pi_username"]
+        # try to update PI
+        try:
+            new_pi = User.objects.get(username=project_pi_username)
+            if new_pi.pi_eligibility().lower() != "eligible":
+                raise UserNotPIEligible()
+            ProjectAllocationMapper.update_project_pi(project_id, project_pi_username)
+            form = EditPIForm(request.POST)
+            messages.success(
+                request,
+                "Update successful! Click <a href='{}'>here</a> to reload".format(
```

Comment:
I don't _think_ this is vulnerable to injection, but the user may have control over `request.path`. Probably not much control as it must have some set of values to even be running in this function. Do we need this "click here to reload" thing at all, do you think?

Reply:
(given that this is only available to super users it's also less of an attack vector IMO)